host.go

```diff
@@ -108,40 +128,117 @@ func (h *gpbftRunner) ValidateMessage(msg *gpbft.GMessage) (gpbft.ValidatedMessa
+		}
+		baseTsk = ts.Key()
+	} else {
+		cert, err := h.client.certstore.Get(h.runningCtx, instance-1)
+		if err != nil {
+			return nil, nil, xerrors.Errorf("getting cert for previous instance(%d): %w", instance-1, err)
+		}
+		baseTsk = cert.ECChain.Head().Key
+	}
 
-	pt, _, err := h.GetCommitteeForInstance(0)
+	baseTs, err := h.client.ec.GetTipset(h.runningCtx, baseTsk)
 	if err != nil {
-		return nil, nil, xerrors.Errorf("getting power table: %w", err)
+		return nil, nil, xerrors.Errorf("getting base TS: %w", err)
+	}
+	headTs, err := h.client.ec.GetHead(h.runningCtx)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("getting head TS: %w", err)
+	}
+
+	collectedChain, err := h.collectChain(baseTs, headTs)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("collecting chain: %w", err)
+	}
+
+	base := gpbft.TipSet{
+		Epoch: baseTs.Epoch(),
+		Key:   baseTs.Key(),
+	}
+	pte, err := h.client.ec.GetPowerTable(h.runningCtx, baseTs.Key())
+	if err != nil {
+		return nil, nil, xerrors.Errorf("getting power table for base: %w", err)
+	}
+	base.PowerTable, err = certs.MakePowerTableCID(pte)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("computing powertable CID for base: %w", err)
 	}
-	ptCid, err := certs.MakePowerTableCID(pt.Entries)
+
+	suffix := make([]gpbft.TipSet, min(gpbft.CHAIN_MAX_LEN-1, len(collectedChain))) // -1 because of base
+	for i := 0; i < len(suffix) && i < len(collectedChain); i++ {
+		suffix[i].Key = collectedChain[i].Key()
+		suffix[i].Epoch = collectedChain[i].Epoch()
+
+		pte, err = h.client.ec.GetPowerTable(h.runningCtx, suffix[i].Key)
+		if err != nil {
+			return nil, nil, xerrors.Errorf("getting power table for suffix %d: %w", i, err)
+		}
+		suffix[i].PowerTable, err = certs.MakePowerTableCID(pte)
+		if err != nil {
+			return nil, nil, xerrors.Errorf("computing powertable CID for base: %w", err)
+		}
+	}
+	chain, err := gpbft.NewChain(base, suffix...)
 	if err != nil {
-		return nil, nil, xerrors.Errorf("computing power table CID: %w", err)
+		return nil, nil, xerrors.Errorf("making new chain: %w", err)
 	}
 
-	ts := sim.NewTipSetGenerator(1)
-	chain, err := gpbft.NewChain(
-		gpbft.TipSet{Epoch: 0, Key: ts.Sample(), PowerTable: ptCid},
-		gpbft.TipSet{Epoch: 1, Key: ts.Sample(), PowerTable: ptCid},
-	)
+	var supplData gpbft.SupplementalData
+	pt, _, err := h.GetCommitteeForInstance(instance + 1)
 	if err != nil {
-		return nil, nil, xerrors.Errorf("geenrating chain: %w", err)
+		return nil, nil, xerrors.Errorf("getting commite for %d: %w", instance+1, err)
 	}
-	sd := &gpbft.SupplementalData{
-		PowerTable: ptCid,
+
+	supplData.PowerTable, err = certs.MakePowerTableCID(pt.Entries)
+	if err != nil {
+		return nil, nil, xerrors.Errorf("making power table cid for supplemental data: %w", err)
 	}
 
-	// TODO: use lookback to return the correct next power table commitment and commitments hash.
-	return sd, chain, nil
+	return &supplData, chain, nil
```

Comment:
hm. Now I see why you wanted to put supplemental data in `GetCommitteeForIInstance`